src/components/TableHead.js

```diff
@@ -19,115 +19,133 @@ const defaultHeadStyles = theme => ({
       display: 'none',
     },
   },
-});
+}), {name: 'MUIDataTableHead'});
 
-class TableHead extends React.Component {
-  componentDidMount() {
-    this.props.handleHeadUpdateRef(this.handleUpdateCheck);
-  }
+function TableHead(props) {
+  
+  const[dragging, setDragging] = useState(false);
 
-  handleToggleColumn = index => {
-    this.props.toggleSort(index);
-  };
+  const {
+    columns,
+    count,
+    options,
+    data,
+    setCellRef,
+    selectedRows,
+    expandedRows,
+    reorderColumns,
+    updateColumnOrder,
+    columnOrder,
+    headCellRefs,
+    timers,
+    toggleSort,
+    tableRef,
+    sortOrder = {},
+    components = {},
+  } = props;
```

Comment:
It would be better to apply PropTypes. 😄

Reply:
Looks like this component never had any, I'll add some in for it - however, I'll probably do it last just because things may change.